odoo/addons/openupgrade_records/lib/compare.py

```diff
@@ -287,11 +287,22 @@ def compare_xml_sets(old_records, new_records):
 
     # direct match
     match_fields = ['module', 'model', 'name']
+    modified_records = []
     for column in copy.copy(old_records):
         found = search(column, new_records, match_fields)
         if found:
             old_records.remove(column)
             new_records.remove(found)
+            if column['domain'] != found['domain'] and \
```

Comment:
Can't all these ifs be simplified a bit? And why you need to do something similar several times? Or at least they can be extracted to a method.

Reply:
ok